src/source/URLSourceProvider.java

```diff
@@ -0,0 +1,54 @@
+package source;
+
+import java.io.*;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+/**
+ * Implementation for loading content from specified URL.<br/>
+ * Valid paths to load are http://someurl.com, https://secureurl.com, ftp://frpurl.com etc.
+ */
+public class URLSourceProvider implements SourceProvider {
+
+    @Override
+    public boolean isAllowed(String pathToSource) {
+        try {
+            new URL(pathToSource).openStream();
+            return true;
+        } catch (IOException e) {
+            return false;
+        }
+    }
+
+    @Override
+    public String load(String pathToSource) throws IOException {
+        HttpURLConnection connection = (HttpURLConnection) new URL(pathToSource).openConnection();
+        int responseCode = connection.getResponseCode();
+
+        if (responseCode != 200) {
+            responseCodeHandling(responseCode);
+        }
+
+        String interLine, finalLine = "";
+        try (BufferedReader is = new BufferedReader(new InputStreamReader(connection.getInputStream()))) {
+            while ((interLine = is.readLine()) != null) {
+                finalLine += interLine;
+            }
+        }
+
+        return finalLine;
+    }
+
+    private void responseCodeHandling(int code) throws IOException {
+        switch (code) {
+            case 401: throw new IOException("Invalid API key");
```

Comment:
You can create enum with codes and messages, to keep enumeration of response codes in one common place.